master/internal/api_checkpoint.go

```diff
@@ -370,7 +370,7 @@ func (a *apiServer) CheckpointsRemoveFiles(
 		go func() {
 			err = runCheckpointGCTask(
 				a.m.system, a.m.rm, a.m.db, taskID, jobID, jobSubmissionTime, taskSpec, exps[i].ID,
-				exps[i].Config, checkpointUUIDs, req.CheckpointGlobs, false, agentUserGroup, curUser,
+				exps[i].Config, checkpointUUIDs, req.CheckpointGlobs, false, false, agentUserGroup, curUser,
```

Comment:
the second false should be true

We need patch checkpoint to update our database state especially for remove checkpoint files. This is the endpoint that we added patchCheckpoints to support the behavior.